testah-junit/src/main/java/org/testah/framework/dto/TestPlanAnnotationDto.java

```diff
@@ -0,0 +1,137 @@
+import org.testah.framework.annotations.TestPlan;
+import org.testah.framework.annotations.TestPlanJUnit5;
+
+import java.lang.annotation.Annotation;
+import java.util.Arrays;
+
+public class TestPlanAnnotationDto {
+
+    private int id = -1;
+    private String name = "";
+    private String description = "";
+    private String[] relatedLinks = new String[]{};
+    private String[] relatedIds = new String[]{};
+    private String[] tags = {};
+    private TestType testType = TestType.DEFAULT;
+    private String[] components = {};
+    private String[] platforms = {};
+    private String[] devices = {};
+    private String[] runTypes = {};
+    private String owner = "";
+
+    public TestPlanAnnotationDto() {
+
+    }
+
+    private TestPlanAnnotationDto(TestPlan testPlan) {
+        this.id = testPlan.id();
+        this.name = testPlan.name();
+        this.description = testPlan.description();
+        this.relatedLinks = testPlan.relatedLinks();
+        this.relatedIds = testPlan.relatedIds();
+        this.testType = testPlan.testType();
+        this.tags = testPlan.tags();
+        this.components = testPlan.components();
+        this.platforms = testPlan.platforms();
+        this.devices = testPlan.devices();
+        this.runTypes = testPlan.runTypes();
+        this.owner = testPlan.owner();
+    }
+
+    private TestPlanAnnotationDto(TestPlanJUnit5 testPlan) {
+        this.id = testPlan.id();
+        this.name = testPlan.name();
+        this.description = testPlan.description();
+        this.relatedLinks = testPlan.relatedLinks();
+        this.relatedIds = testPlan.relatedIds();
+        this.testType = testPlan.testType();
+        this.tags = testPlan.tags();
+        this.components = testPlan.components();
+        this.platforms = testPlan.platforms();
+        this.devices = testPlan.devices();
+        this.runTypes = testPlan.runTypes();
+        this.owner = testPlan.owner();
+    }
+
+    /**
+     * Create TestPlanAnnotationDto from a testClass.
+     * @param testClass testPlan Class.
+     * @return return new TestPlanAnnotationDto.
+     */
+    public static TestPlanAnnotationDto create(final Class testClass) {
+        if (testClass != null) {
+            return create(testClass.getAnnotations());
+        }
+        return null;
+    }
+
+    /**
+     * Create TestPlanAnnotationDto from a testPlan annotation.
+     * @param testPlans testPlan annotation.
+     * @return return new TestPlanAnnotationDto.
+     */
+    public static TestPlanAnnotationDto create(Annotation... testPlans) {
+        if (testPlans != null) {
+            for (Annotation testPlan : testPlans) {
+                if (testPlan instanceof TestPlan) {
+                    return new TestPlanAnnotationDto((TestPlan) testPlan);
+                } else if (testPlan instanceof TestPlanJUnit5) {
+                    return new TestPlanAnnotationDto((TestPlanJUnit5) testPlan);
+                }
+            }
```

Comment:
else not implemented exception, better than returning null?

Reply:
I agree, but has to be just for junit 5, as junit 4 implementation allowed it, although not by desire.